ansible/roles/docker/tasks/docker-compose-install-v2.yml

```diff
@@ -1,15 +1,20 @@
 ---
 # The Ubuntu package of Docker Compose cannot be used as it pulls outdated
 # depdendencies which clash with the Docker Engine installation
-- name: Download the Docker Compose executable
+- name: Remove Docker Compose V1
+  ansible.builtin.file:
+    path: /usr/bin/docker-compose
+    state: absent
+
+- name: Download the Docker Compose v2 executable
   ansible.builtin.get_url:
-    url: "{{ docker_compose_url }}"
+    url: "{{ docker_compose_v2_url }}"
     dest: "{{ docker_cli_plugins_dir }}/docker-compose"
     owner: root
     group: root
     mode: 0755
-    checksum: "sha256:{{ lookup('url', docker_compose_digest_url) | regex_search('[0-9a-f]{64}') }}"
+    checksum: "sha256:{{ lookup('url', docker_compose_v2_digest_url) | regex_search('[0-9a-f]{64}') }}"
 
```

Comment:
```suggestion
  become: true

```

we probably need a `become: true` at the end of this task, if we want root to own the file?

Reply:
I haven't included `become: true` as I wanted to be consistent with the rest of the role. The other two tasks in this role don't specify it but require root as well. The current setup works because we have a number of `become: true` defined at the play level in `site.yml`.

If you prefer specifying `become` at the role level instead of the play level, it's ok by me, but in that case perhaps we should do it for the whole role.

Reply:
I would prefer this. This way we do not get unpleasant surprises when accidentally forgetting the become: true on the play-level. Also this way only the tasks that actually need privilege escalation get it.

Reply:
Added in `tasks/main.yml`.